ibis/expr/window.py

```diff
@@ -85,11 +72,13 @@ def _validate_frame(self):
             else:
                 assert start < end
         else:
-            if has_p and self.preceding < 0:
-                raise com.IbisInputError('Window offset must be positive')
+            if not isinstance(self.preceding, ir.Expr):
+                if has_p and self.preceding < 0:
+                    raise com.IbisInputError('Window offset must be positive')
 
-            if has_f and self.following < 0:
-                raise com.IbisInputError('Window offset must be positive')
+            if not isinstance(self.following, ir.Expr):
+                if has_f and self.following < 0:
+                    raise com.IbisInputError('Window offset must be positive')
```

Comment:
Different conditions are raising the same error, perhaps more informative msgs?

Reply:
Good point.